### ISSUE TITLE
keyframe indextable sometimes broken

### ISSUE THREAD

**Issue by donkeykung** [fnordware/AdobeWebM#35]
I noticed that some webm's created with premiere are kinda weird. When opened in vlc I sometimes can't fast forward properly. Is there a way to fix it?

**Comment by donkeykung** [comment]
Here is the messed up webm.

https://mega.nz/#!V0ZFSI7Z!-64amAAUUL3swPG9M3_NpG-hBuOrGAALBg24HbYOGCM

when I turn the same video into a webm using the freeware xmedia it doesn't happen

https://mega.nz/#!xhQUxZJI!9VoeDD8EP4jUYl3A3BOEU0wAkKOnXYJsjdzG0xIeBuM

And another thing, the download on this page: http://www.fnordware.com/WebM/ isn't it out of date and should be updated?

**Comment by fnordware** [comment]
I have to imagine that problems with scrubbing are due to long gaps between keyframes. I use the default, which might not make many keyframes if your video isn't rapidly changing. You can use the --kf-max-dist=[your num] to put a cap on it, which should help. Xmedia maybe uses a different default.

I've considered exposing this setting as a regular control.

**Comment by fnordware** [comment]
The plug-in at http://www.fnordware.com/WebM/ was up to date, but the date listed was not. I've fixed it.

**Comment by donkeykung** [comment]
Let me try it this way, I create webm's and then share them on some websites. The problem I have now is that some webm's, after  they have been upoaded end up having no thumbnail.
![2016-07-14_17-43-08](https://cloud.githubusercontent.com/assets/19196062/16845876/b8636af8-49ea-11e6-92cd-0bcfbec855cb.png) 

And this only ever happens with webm's created by Adobe Premiere.

For example here are 2 webm's created by other people
![2016-07-14_17-46-49](https://cloud.githubusercontent.com/assets/19196062/16846025/32528664-49eb-11e6-9061-00884d15c0cc.png)

They do have thumbnails while I get only that weird white arrow, which is kinda crappy because people always skip those and my posts get ignored because of that.

I have asked one of them if they know what could cause this and one guy said "Yeah, the premiere one seems to be slightly corrupted. I think there's something wrong with the keyframe indextable. Most players seem to have at least some problem seeking through the content, or flip out when you do. It would explain the missing thumbnail, because that's simply a grab of the centermost keyframe, which in this case, can't be found correctly.". Idk if what he said is true, I just assumed it is.

**Comment by fnordware** [comment]
I downloaded your files and the Premiere one does seem to have some scrubbing issues in Chrome while the other doesn't. I'm not really sure why—will show it to the Google people.

I'm using Google's libwebm muxing library, while FFmpeg has its own. Maybe it's doing something wrong or maybe the site is doing something specific to the FFmpeg method. Both files appear to have a valid set of Cues.

**Comment by fnordware** [comment]
BTW, both files are able to scrub in VLC, although curiously the one that can scrub in Chrome does a weird little stutter when you skip in VLC.

**Comment by fnordware** [comment]
Filed the issue with Google here:

https://bugs.chromium.org/p/webm/issues/detail?id=1257

**Comment by fnordware** [comment]
We got a response, and it's to set a maximum keyframe distance, as I said above.

Perhaps I should expose that parameter in a proper control or set a different default internally.

**Comment by donkeykung** [comment]
Okay cool, thanks for the help.

> If random seeking is a priority for the encoded output my personal preference is somewhere in the 1-10x frame rate range, so for 30 frames/second input, 30-300.

Is there currently a parameter to set the max keyframe? Going trough the manual right now, is it the arnr maxframes one? Also, is it basically fps*10? E.g. if the video has 24 frames per sec I set it to 240?

**Comment by fnordware** [comment]
Yeah, the max distance is set in terms of the number of frames. In the custom args field for a keyframe every 10 seconds at 24fps you'd want:

--kf-max-dist=240
